src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,187 @@
+            result.append("No matching tasks found.");
+        } else {
+            result.append("Here are the matching tasks in your list:");
+            int taskCount = 1;
+            for (Task task : foundTasks) {
+                result.append("\n ").append(taskCount).append(".").append(task);
+                taskCount++;
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Generates a formatted string containing the list of tasks and returns the result.
+     *
+     * @param tasks The ArrayList of tasks to be displayed.
+     * @return A formatted string containing the list of tasks.
+     */
+    public String showAllTasks(ArrayList<Task> tasks) {
+        StringBuilder result = new StringBuilder();
+        if (tasks.isEmpty()) {
+            result.append("There are no tasks in the list.");
+        } else {
+            result.append("Here are the tasks in your list:");
+            int taskCount = 1;
+            for (Task task : tasks) {
+                result.append("\n ").append(taskCount).append(".").append(task);
+                taskCount++;
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Generates a formatted string for marking a task as not done and returns the result.
+     *
+     * @param task The task to be marked as not done.
+     * @return A formatted string confirming the task has been marked as not done.
+     */
+    public String showUnmarkTaskMessage(Task task) {
+        return "OK, I've marked this task as not done yet:\n" + "   " + task;
+    }
+
+    /**
+     * Generates a formatted string for marking a task as done and returns the result.
+     *
+     * @param task The task to be marked as done.
+     * @return A formatted string confirming the task has been marked as done.
+     */
+    public String showMarkTaskMessage(Task task) {
+        return "Nice! I've marked this task as done:\n" + "   " + task;
+    }
+
+    /**
+     * Generates a formatted string for adding a task and returns the result.
+     *
+     * @param task The task that has been added.
+     * @param taskCount The current count of tasks in the list.
+     * @return A formatted string confirming the addition of the task.
+     */
+    public String showAddTaskMessage(Task task, int taskCount) {
+        return "Got it. I've added this task:\n" + "   " + task + "\n"
+                + " Now you have " + taskCount + " task(s) in the list.";
+    }
+
+    /**
+     * Generates a formatted string for deleting a task and returns the result.
+     *
+     * @param task The task that has been deleted.
+     * @param taskCount The current count of tasks in the list.
+     * @return A formatted string confirming the deletion of the task.
+     */
+    public String showDeleteTaskMessage(Task task, int taskCount) {
+        return "Noted. I've removed this task:\n" + "   " + task + "\n"
+                + " Now you have " + taskCount + " task(s) in the list.";
+    }
```

Comment:
Maybe you could consider storing these as format strings instead of methods, can refer to the way AB2 does it :)